### PR TITLE
Fix filters for meta columns

### LINKED ISSUE #998: Tables appear empty (No entries) after update from 0.6.6 to  0.7.0 official (via App Store)
### Steps to reproduce

Steps to reproduce
    Have Tables 0.6.6 installed
    Have a table with data
    Upgrade to Tables 0.7.0 official (via App Store)

### Expected behavior

    Upgrade works fine
    Tables still have their table

### Actual behavior

 Tables appear empty (No entries/rows are displayed in the table)

### Tables app version

0.7.0 official

### Browser

Firefox latest, Chrome latest

### Client operating system

Linux OpenSuse

### Operating system

 Linux 4.18.0-513.18.2.el8_9.x86_64 x86_64

### Web server

None

### PHP engine version

PHP 8.2

### Database

MySQL

### Additional info

Nextcloud 28.0.4

Systemprotocol: InternalError No column found to build filter with for id -2

[tables] Fehler: No column found to build filter with for id -2
    GET /index.php/apps/tables/table
    von 178.13.21.33 von MaxMuster um 25.04.2024, 10:56:19

```
{
  "reqId": "ZioaswDdI8MDFKhqcyDKKAAAAAY",
  "level": 3,
  "time": "2024-04-25T08:56:19+00:00",
  "remoteAddr": "178.13.21.33",
  "user": "MaxMuster",
  "app": "tables",
  "method": "GET",
  "url": "/index.php/apps/tables/table",
  "message": "No column found to build filter with for id -2",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64; rv:124.0) Gecko/20100101 Firefox/124.0",
  "version": "28.0.4.1",
  "exception": {
    "Exception": "OCA\\Tables\\Errors\\InternalError",
    "Message": "No column found to build filter with for id -2",
    "Code": 0,
    "Trace": [
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Db/Row2Mapper.php",
        "line": 263,
        "function": "getFilter",
        "class": "OCA\\Tables\\Db\\Row2Mapper",
        "type": "->",
        "args": [
          [
            "OC\\DB\\QueryBuilder\\QueryBuilder"
          ],
          [
            [
              "Max Muster",
              -2,
              "is-equal"
            ]
          ]
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Db/Row2Mapper.php",
        "line": 241,
        "function": "getFilterGroups",
        "class": "OCA\\Tables\\Db\\Row2Mapper",
        "type": "->",
        "args": [
          [
            "OC\\DB\\QueryBuilder\\QueryBuilder"
          ],
          [
            [
              [
                "Max Muster",
                -2,
                "is-equal"
              ]
            ]
          ]
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Db/Row2Mapper.php",
        "line": 143,
        "function": "addFilterToQuery",
        "class": "OCA\\Tables\\Db\\Row2Mapper",
        "type": "->",
        "args": [
          [
            "OC\\DB\\QueryBuilder\\QueryBuilder"
          ],
          [
            [
              [
                "Max Muster",
                -2,
                "is-equal"
              ]
            ]
          ],
          "MaxMuster"
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Db/Row2Mapper.php",
        "line": 827,
        "function": "getWantedRowIds",
        "class": "OCA\\Tables\\Db\\Row2Mapper",
        "type": "->",
        "args": [
          "MaxMuster",
          41,
          [
            [
              [
                "@my-name",
                -2,
                "is-equal"
              ]
            ]
          ]
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Service/RowService.php",
        "line": 535,
        "function": "countRowsForView",
        "class": "OCA\\Tables\\Db\\Row2Mapper",
        "type": "->",
        "args": [
          [
            "OCA\\Tables\\Db\\View",
            6
          ],
          "MaxMuster",
          [
            [
              "OCA\\Tables\\Db\\Column",
              168
            ],
            [
              "OCA\\Tables\\Db\\Column",
              169
            ],
            [
              "OCA\\Tables\\Db\\Column",
              177
            ],
            [
              "OCA\\Tables\\Db\\Column",
              198
            ],
            [
              "OCA\\Tables\\Db\\Column",
              199
            ],
            "And 9 more entries, set log level to debug to see all entries"
          ]
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Service/ViewService.php",
        "line": 413,
        "function": "getViewRowsCount",
        "class": "OCA\\Tables\\Service\\RowService",
        "type": "->",
        "args": [
          [
            "OCA\\Tables\\Db\\View",
            6
          ],
          "MaxMuster"
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Service/ViewService.php",
        "line": 90,
        "function": "enhanceView",
        "class": "OCA\\Tables\\Service\\ViewService",
        "type": "->",
        "args": [
          [
            "OCA\\Tables\\Db\\View",
            6
          ],
          "MaxMuster"
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Service/TableService.php",
        "line": 227,
        "function": "findAll",
        "class": "OCA\\Tables\\Service\\ViewService",
        "type": "->",
        "args": [
          [
            "OCA\\Tables\\Db\\Table",
            41
          ]
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Service/TableService.php",
        "line": 149,
        "function": "enhanceTable",
        "class": "OCA\\Tables\\Service\\TableService",
        "type": "->",
        "args": [
          [
            "OCA\\Tables\\Db\\Table",
            41
          ],
          "MaxMuster"
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Controller/TableController.php",
        "line": 39,
        "function": "findAll",
        "class": "OCA\\Tables\\Service\\TableService",
        "type": "->",
        "args": [
          "MaxMuster"
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Controller/Errors.php",
        "line": 16,
        "function": "OCA\\Tables\\Controller\\{closure}",
        "class": "OCA\\Tables\\Controller\\TableController",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "/home/www/nextcloud/apps/tables/lib/Controller/TableController.php",
        "line": 38,
        "function": "handleError",
        "class": "OCA\\Tables\\Controller\\TableController",
        "type": "->",
        "args": [
          [
            "Closure"
          ]
        ]
      },
      {
        "file": "/home/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 230,
        "function": "index",
        "class": "OCA\\Tables\\Controller\\TableController",
        "type": "->",
        "args": []
      },
      {
        "file": "/home/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php",
        "line": 137,
        "function": "executeController",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          [
            "OCA\\Tables\\Controller\\TableController"
          ],
          "index"
        ]
      },
      {
        "file": "/home/www/nextcloud/lib/private/AppFramework/App.php",
        "line": 184,
        "function": "dispatch",
        "class": "OC\\AppFramework\\Http\\Dispatcher",
        "type": "->",
        "args": [
          [
            "OCA\\Tables\\Controller\\TableController"
          ],
          "index"
        ]
      },
      {
        "file": "/home/www/nextcloud/lib/private/Route/Router.php",
        "line": 315,
        "function": "main",
        "class": "OC\\AppFramework\\App",
        "type": "::",
        "args": [
          "OCA\\Tables\\Controller\\TableController",
          "index",
          [
            "OC\\AppFramework\\DependencyInjection\\DIContainer"
          ],
          [
            "tables.table.index"
          ]
        ]
      },
      {
        "file": "/home/www/nextcloud/lib/base.php",
        "line": 1069,
        "function": "match",
        "class": "OC\\Route\\Router",
        "type": "->",
        "args": [
          "/apps/tables/table"
        ]
      },
      {
        "file": "/home/www/nextcloud/index.php",
        "line": 39,
        "function": "handleRequest",
        "class": "OC",
        "type": "::",
        "args": []
      }
    ],
    "File": "/home/www/nextcloud/apps/tables/lib/Db/Row2Mapper.php",
    "Line": 276,
    "message": "No column found to build filter with for id -2",
    "exception": [],
    "CustomMessage": "No column found to build filter with for id -2"
  },
  "id": "662a1bbc0c62f"
}


```

### PR DESCRIPTION
**Pull request by juliushaertl**
- **fix: Avoid failing if a view filter contains a meta column**
- **chore: Move meta column identifiers to constants**

Likely fixes #998

### CODE REVIEW

**Comment by enjeck** [comment]
Cool, this probably also fixes this https://github.com/nextcloud/tables/pull/979. We had been getting similar error messages for the vacation template, and I couldn't figure out how to properly fix it

**Comment by juliushaertl** [comment]
> Cool, this probably also fixes this #979. We had been getting similar error messages for the vacation template, and I couldn't figure out how to properly fix it

Oh indeed, i also didn't have this on my radar when reviewing your PR. Now I gave this a try with reverting your fix, the vacation request template seems to work properly again with this PR.

**Comment by juliushaertl** [comment]
Pushed the revert and two more commits to fix an inconsistency compared to previous versions. Also moved frontend code to use the same constants

**Comment by blizzz** [comment]
> Cool, this probably also fixes this #979. We had been getting similar error messages for the vacation template, and I couldn't figure out how to properly fix it

And i had a foul feeling not seeing constants there.

File: lib/Service/TableTemplateService.php [hunk 1]
```diff
@@ -469,8 +469,8 @@ private function makeVacationRequests(Table $table):void {
 				'title' => $this->l->t('Create Vacation Request'),
 				'emoji' => '️➕',
 				'columns' => json_encode([$columns['employee']->getId(), $columns['from']->getId(), $columns['to']->getId(), $columns['workingDays']->getId(), $columns['dateRequest']->getId()]),
-				'sort' => json_encode([["columnId" => $columns['from']->getId(), "mode" => "ASC"]]),
-				'filter' => json_encode([[["columnId" => $columns['employee']->getId(), "operator" => "is-equal", "value" => "@my-name"], ["columnId" => $columns['approved']->getId(), "operator" => "is-empty", "value" => ""]]]),
+				'sort' => json_encode([["columnId" => -5, "mode" => "ASC"]]),
```
Comment (blizzz) [inline lib/Service/TableTemplateService.php:472]: Constants not used in this file (also the changes below)

**Review by blizzz — CHANGES_REQUESTED** [review]
more usage of constants

File: lib/Service/TableTemplateService.php (hunk 1 above)
Comment (juliushaertl) [reply lib/Service/TableTemplateService.php]: Pushed another commit for two more places as well.

**Pull request merged**